src/test/credentials/provider/ssoAccessTokenProvider.test.ts

```diff
@@ -0,0 +1,230 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as AWS from 'aws-sdk'
+import * as sinon from 'sinon'
+import { SSOOIDC } from 'aws-sdk'
+import { DiskCache } from '../../../credentials/sso/diskCache'
+import { SsoAccessTokenProvider } from '../../../credentials/sso/ssoAccessTokenProvider'
+import { CreateTokenResponse, StartDeviceAuthorizationResponse } from 'aws-sdk/clients/ssooidc'
+import { SsoClientRegistration } from '../../../credentials/sso/ssoClientRegistration'
+import { SsoAccessToken } from '../../../credentials/sso/ssoAccessToken'
+
+describe('SsoAccessTokenProvider', () => {
+    let sandbox = sinon.createSandbox()
+
+    const ssoRegion = 'fakeRegion'
+    const ssoUrl = 'fakeUrl'
+    const ssoOidcClient = new SSOOIDC({ region: 'us-west-2' })
+    const cache = new DiskCache()
+    const sut = new SsoAccessTokenProvider(ssoRegion, ssoUrl, ssoOidcClient, cache)
+
+    const HOUR_IN_MS = 3600000
+
+    const validAccessToken = {
```

Comment:
Nit: combine with the other `validAccessToken` definition and export from a common location?